go/vt/schemadiff/schema_diff.go

```diff
@@ -343,3 +345,24 @@ func (d *SchemaDiff) OrderedDiffs(ctx context.Context) ([]EntityDiff, error) {
 	}
 	return orderedDiffs, nil
 }
+
+// CapableOfInstantDDL returns `true` if all diffs are capable of instant DDL, or are otherwise trivially
+// instantaneously applicable (such as `CREATE TABLE` or `ALTER VIEW`). The answer essentially indicates whether
+// the entire set of changes can be applied as an immediate operation.
+func (d *SchemaDiff) CapableOfInstantDDL(ctx context.Context, capableOf capabilities.CapableOf) (bool, error) {
+	if capableOf == nil {
+		return false, nil
+	}
+	var errs error
+	allCapable := true
+	for _, diff := range d.UnorderedDiffs() {
+		capable, err := diffCapableOfInstantDDL(diff, capableOf)
+		if err != nil {
+			errs = errors.Join(errs, err)
```

Comment:
Just noting that this uses wrapping (which generally means the outer error was caused by the inner). I'm not sure if that's what we want here?

Reply:
To the best of my knowledge, `errors.Join` is not doing wrapping. It implements a slices of `err` objects, most of which are inaccessible unless you explicitly cast your `err` to an `Unwrap() []error` interface, like so: 

https://github.com/vitessio/vitess/blob/bd9e53461694244a08ca1cf7b2d2deeccf0348d9/go/errors/errors.go#L20-L33

Concatenating errors is a paradigm we use throughout `schemadiff`, with the explicit intention of returning as much information about diff/problems in one go.

Reply:
BTW I realize the above code reads `Wrapped` and `Unwrap`, but I think it's not in the same way you meant it (meaning it's not hierarchical parent->child, but flat). Possibly we should rename those function and type.